test/ranking.jl

```diff
@@ -37,3 +37,76 @@ s = ["c", "a", "b", "d", "d", "b", "e", "d"] # s is a vector of strings ordered
 @test tiedrank(s) == tiedrank(x)
 @test tiedrank(x, rev = true) == tiedrank(-x)
 @test tiedrank(x, lt = (x, y) -> isless(y, x)) == tiedrank(-x)
+
+@testset "quantilerank and percentrank" begin
+    @testset "value as number and array" begin
+        @testset ":inc and :exc" begin
+            v1 = [1, 1, 1, 2, 3, 4, 8, 11, 12, 13]
+            v2 = [1, 2, 3, 6, 6, 6, 7, 8, 9]
+            v3 = [1, 2, 4, 3, 4]
+            v4 = [1, 2, 1, 3, 4]
+            @test quantilerank(v1, 2, method=:inc) == 1/3
+            @test quantilerank(v1, 4, method=:inc) == 5/9
+            @test quantilerank(v1, 8, method=:inc) == 2/3
+            @test quantilerank(v1, 5, method=:inc) == 7/12
+            @test percentrank(v1,  2, method=:inc) == 100 * (1/3)
+            @test percentrank(v1,  4, method=:inc) == 100 * (5/9)
+            @test percentrank(v1,  8, method=:inc) == 100 * (2/3)
+            @test percentrank(v1,  5, method=:inc) == 100 * (7/12)
+            @test quantilerank(v2, 7, method=:exc)    == 0.7
+            @test quantilerank(v2, 5.43, method=:exc) == 0.381
+            @test percentrank(v2, 7, method=:exc)     == 70.0
+            @test percentrank(v2, 5.43, method=:exc)  == 38.1
+            @test quantilerank(v3, 4, method=:exc) == 6/9
+            @test quantilerank(v3, 4, method=:inc) == 3/4
+            @test quantilerank(v4, 1, method=:exc) == 1/6
+            @test quantilerank(v4, -100, method=:inc) == 0.0
+            @test quantilerank(v4,  100, method=:inc) == 1.0
+            @test quantilerank(v4, -100, method=:exc) == 0.0
+            @test quantilerank(v4,  100, method=:exc) == 1.0
+        end
+        @testset ":compete" begin
+            v = [0, 0, 1, 1, 2, 2, 2, 2, 4, 4]
+            @test quantilerank(v, 1, method=:compete) == 2/9
+            @test quantilerank(v, 2, method=:compete) == 4/9
+            @test quantilerank(v, 4, method=:compete) == 8/9
+            @test percentrank(v,  1, method=:compete) == 100 * (2/9)
+            @test percentrank(v,  2, method=:compete) == 100 * (4/9)
+            @test percentrank(v,  4, method=:compete) == 100 * (8/9)
+            @test quantilerank(v, -100, method=:compete) == 0.0
+            @test quantilerank(v,  100, method=:compete) == 1.0
+        end
+        @testset ":strict, :weak and :tied" begin
+            v = [7, 8, 2, 1, 3, 4, 5, 4, 6, 9]
+            for (method, res1, res2) in [(:tied, .4, [.4, .85]),
+                                        (:strict, .3, [.3, .8]),
+                                        (:weak, .5, [.5, .9])]
+                @test percentrank(v,  4, method=method) == res1 * 100
+                @test quantilerank(v, 4, method=method) == res1
+                @test percentrank.(Ref(v),  [4, 8], method=method) == res2 * 100
+                @test quantilerank.(Ref(v), [4, 8], method=method) == res2
```

Comment:
This isn't really useful as it just tests `broadcast` in Base.
```suggestion
```

Reply:
Done in commit 3e30dd1.